src/Compilers/CSharp/Portable/Lowering/AsyncRewriter/AwaitExpressionSpiller.cs

```diff
@@ -444,6 +452,8 @@ private BoundStatement UpdateStatement(BoundSpillSequenceBuilder builder, BoundS
             bool forceSpill = false,
             bool sideEffectsOnly = false)
         {
+            Debug.Assert(refKinds.IsDefaultOrEmpty || refKinds.Length == args.Length);
```

Comment:
not sure. the existing code seems to have assumed that either empty or default may be used to mean "All None". One way to find out..

---
In reply to: [127523835](https://github.com/dotnet/roslyn/pull/20883#discussion_r127523835) [](ancestors = 127523835)